explorer/src/api/index.ts

```diff
@@ -4,17 +4,64 @@ import {
   VALIDATORS_API,
   BLOCK_API,
   COUNTRY_DATA_API,
+  DELEGATIONS_API,
+  MIXNODE_STATS_API,
+  MIXNODE_PING,
+  UPTIME_STORY_API,
 } from './constants';
 
-import { MixNodeResponse, GatewayResponse, ValidatorsResponse, CountryDataResponse } from '../typeDefs/explorer-api'
+import { 
+  MixNodeResponse,
+  GatewayResponse,
+  ValidatorsResponse,
+  CountryDataResponse,
+  MixNodeResponseItem,
+  DelegationsResponse,
+  StatsResponse,
+  StatusResponse,
+  UptimeStoryResponse,
+} from '../typeDefs/explorer-api'
+
+function getFromCache(key: string) {
+  const ts = Number(localStorage.getItem('ts'));
+  const hasExpired = Date.now() - ts > 200000;
+  const curr = localStorage.getItem(key);
+  if (curr && !hasExpired) {
+    return JSON.parse(curr);
+  } else {
+    return undefined;
+  }
+}
+
+function storeInCache(key: string, data: any) {
+  localStorage.setItem(key, data);
+}
+
+function clearCache(key: string) {
+  return localStorage.removeItem(key);
+}
```

Comment:
As we seem to have solved the performance issue using pagination with the data grid, we might now need any caching for the mixnode data?

Wait to see what Mark says though.

Reply:
Agree